packages/auth/src/Api/reducer.ts

```diff
@@ -1,168 +1,43 @@
-  AuthState,
-  ForgotPasswordState,
-  ForgotPasswordStep,
-  LoginState,
-  LoginStep,
-  MFAState,
-  MFAStep,
-  ProfileState,
-  SSOState,
-  User,
-} from './interfaces';
-import {
-  IActivateAccount,
-  IChangePassword,
-  IDisableMfa,
-  IForgotPassword,
-  ILogin,
-  ILoginWithMfa,
-  IPostLogin,
-  IPreLogin,
-  IRecoverMFAToken,
-  IResetPassword,
-  ISamlConfiguration,
-  IUserProfile,
-  IVerifyMfa,
-} from '@frontegg/react-core';
-
-export const storeName = 'auth';
-
-export const preloadedState: AuthState = {
-  // routes
-  routes: {
-    authenticatedUrl: '/',
-    loginUrl: '/account/login',
-    logoutUrl: '/account/logout',
-    activateUrl: '/account/activate',
-    forgetPasswordUrl: '/account/forget-password',
-    resetPasswordUrl: '/account/reset-password',
-  },
-  onRedirectTo: () => {},
-
-  isAuthenticated: false,
-  isLoading: true,
-  isSSOAuth: false,
-  user: null,
-
-  loginState: {
-    step: LoginStep.preLogin,
-    loading: false,
-    email: '',
-  },
-
-  activateState: {
-    step: ActivateStep.activating,
-    loading: false,
-  },
-
-  forgetPasswordState: {
-    step: ForgotPasswordStep.forgotPassword,
-    loading: false,
-    email: '',
-  },
-
-  ssoState: {
-    firstLoad: true,
-    loading: true,
-  },
-
-  profileState: {
-    loading: false,
-  },
-  mfaState: {
-    step: MFAStep.verify,
-    loading: false,
-  },
-};
-
-const resetStateByKey = <T>(key: keyof AuthState) => (state: AuthState) => ({ ...state, [key]: preloadedState[key] });
-const typeReducer = <T>(key: keyof AuthState) => (state: AuthState, { payload }: PayloadAction<T>) => ({
-  ...state,
-  [key]: payload,
-});
-const typeReducerForKey = <T>(key: keyof AuthState) => ({
-  prepare: (payload: Partial<T>) => ({ payload }),
-  reducer: (state: AuthState, { payload }: PayloadAction<Partial<T>>) => {
-    return {
-      ...state,
-      [key]: {
-        ...state[key],
-        ...payload,
-      },
-    };
-  },
-});
-
-const { reducer, actions: SliceActions } = createSlice({
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { AuthState, User } from './interfaces';
+import { storeName, typeReducer } from './utils';
+import { initialState } from './initialState';
+import { loginActions, loginStateReducers } from './LoginState';
+import { activateActions, activateStateReducers } from './ActivateState';
+import { forgotPasswordActions, forgotPasswordStateReducers } from './ForgotPasswordState';
+import { ssoActions, ssoStateReducers } from './SSOState';
+import { mfaActions, mfaStateReducers } from './MfaState';
+import { profileActions, profileStateReducers } from './ProfileState';
+import { teamActions, teamStateReducers } from './TeamState';
+
+const { reducer, actions: sliceActions } = createSlice({
   name: storeName,
-  initialState: preloadedState,
+  initialState,
   reducers: {
     setState: (state: AuthState, { payload }: PayloadAction<Partial<AuthState>>) => ({ ...state, ...payload }),
```

Comment:
Why does it call setState?

Reply:
this is hook for other plugins to `setState` inside the **auth plugin**

Reply:
How about setAuthState?

Reply:
I see that type is AuthState, but I think it'll be better

Reply:
it always be under AuthPlugin 
import {authActions} from '@frontegg/react-auth'
authActions.setState()

import {auditsActions} from '@frontegg/react-audits'
auditsActions.setState()

...etc